### PR TITLE
feat: example walletconnect login flow

### PR DESCRIPTION
**Pull request by dmccartney**
This demonstrates the WalletConnect login flow in the example app.

It also simplifies the wallet creation interface.

### CODE REVIEW

File: example/lib/pages/home_page.dart [hunk 1]
```diff
@@ -18,10 +19,16 @@ class HomePage extends HookWidget {
...
   Widget build(BuildContext context) {
+    var me = useMe();
     var conversations = useConversationList();
     var refresher = useConversationsRefresher();
     return Scaffold(
-        appBar: AppBar(),
+        appBar: AppBar(title: AddressChip(address: me), actions: [
+          IconButton(
+            icon: const Icon(Icons.logout),
+            onPressed: () => clearSession(),
```
Comment (elisealix22) [inline example/lib/pages/home_page.dart:29]: ❤️

File: example/lib/session.dart [hunk 2]
```diff
@@ -1,28 +1,59 @@
...
+  await session.reset();
+}
+
+Future<bool> initNewSession(xmtp.Signer wallet) async {
+  try {
+    WidgetsFlutterBinding.ensureInitialized();
+    var prefs = await SharedPreferences.getInstance();
+    var api = xmtp.Api.create(host: 'dev.xmtp.network', isSecure: true);
+    var client = await xmtp.Client.createFromWallet(api, wallet);
+    await prefs.setString("xmtp.keys", client.keys.writeToJson());
```
Comment (elisealix22) [inline example/lib/session.dart:50]: SharedPrefs seems totally reasonable for the time being, but should we use the encrypted https://pub.dev/packages/flutter_secure_storage in the long run? Relating this to SharedPrefs vs [Android Keystore](https://developer.android.com/training/articles/keystore.html) which can be encrypted and prevents malicious extraction from the device.

>This is an advanced security feature that is generally useful only if your requirements are that a compromise of your application process after key generation/import (but not before or during) can't lead to unauthorized uses of the key.

**Review by elisealix22 — APPROVED** [review]
SDK changes look great to me! One q for best practices in the example app around storing keys.

**Comment by nplasterer** [comment]
Just wanted to add that I pulled this down and tested it and everything worked as expected from JS to Android Flutter connected via a rainbow wallet.

photo_2022-12-31 09 47 14

**Pull request merged**